matchbox_server/src/signaling.rs

```diff
@@ -62,19 +69,17 @@ impl State {
         let peer_id = peer.uuid.clone();
         let room = peer.room.clone();
         self.clients.insert(peer.uuid.clone(), peer);
+        let peers = self.rooms.entry(room.clone()).or_default();
 
-        match room {
-            RequestedRoom::Id(room_id) => {
-                let peers = self.id_rooms.entry(room_id).or_default();
-                let ret = peers.iter().cloned().collect();
+        let ret = peers.iter().cloned().collect();
+        match room.next {
+            None => {
                 peers.insert(peer_id);
                 ret
             }
-            RequestedRoom::Next(num_players) => {
-                let peers = self.next_rooms.entry(num_players).or_default();
-                let ret = peers.iter().cloned().collect();
+            Some(num_players) => {
                 if peers.len() == num_players - 1 {
-                    peers.clear() // the room is complete, we can forget about it now
+                    peers.clear(); // the room is complete, we can forget about it now
```

Comment:
Seeing no semicolon troubled me more than it should have.

`clear()` is returning () ; so the entire if clauses returns () and compiler doesn't complain...

But as a reader, I parse the absence of semicolon to be a return of a value, which makes no sense here.